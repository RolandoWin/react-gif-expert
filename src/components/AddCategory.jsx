import { PropTypes } from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        //console.log( target.value );
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
        event.preventDefault();
        if( inputValue.trim().length <= 1 ) return;
        //console.log(inputValue);
        // setCategories(...inputValue
        //setCategories( categories => [ inputValue, ...categories] );
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <>
            <form onSubmit = { (event) => onSubmit(event) } aria-label="form">
                <input 
                    type="text" 
                    placeholder="Buscar Gifs" 
                    value={inputValue}
                    onChange={ (event) => onInputChange(event) }
                />
            </form>
        </>
    )
};

//Hacemos que onNewCategory sea obligatorio
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
