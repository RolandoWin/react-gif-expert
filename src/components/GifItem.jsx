import PropTypes from 'prop-types';

export const GifItem = ({ title, id, url}) => {
    return (
        <div className="card">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}


GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
}

/* Tarea 

1.- AÃ±adir Proptypes
    a.- title obligatorio
    b.- url obligatorio

2.- Evaluar el snapshot
*/