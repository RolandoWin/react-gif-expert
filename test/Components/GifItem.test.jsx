import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'Dragon Ball GIF by Toei Animation';
    const url = 'https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=836777ffnqkzd1rx88yrp4exgkp0bhwq5atnuhclxuxfff0k&amp;rid=giphy.gif&amp;ct=g';
    const id = 'GRSnxyhJnPsaQy9YLn';
    
    test('1.- Debe hacer match con el snapshot', () => {
         
        const { container } = render( <GifItem title={title} url={url} id={id} />);
        expect(container).toMatchSnapshot();

    });

    test('2.- Debe mostrar la imagen del url  y el alt indicado', () => {
        
        render( <GifItem title={title} url={url} id={id} />);
        //screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

    test('3.- Debe mostrar el tÃ­tulo en el componente', () => {
        
        render( <GifItem title={title} url={url} id={id} />);
        expect( screen.getByText( title ) ).toBeTruthy();

    });
    
    
    
});