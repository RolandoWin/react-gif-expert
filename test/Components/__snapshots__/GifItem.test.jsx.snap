// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> 1.- Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball GIF by Toei Animation"
      src="https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=836777ffnqkzd1rx88yrp4exgkp0bhwq5atnuhclxuxfff0k&amp;rid=giphy.gif&amp;ct=g"
    />
    <p>
      Dragon Ball GIF by Toei Animation
    </p>
  </div>
</div>
`;
